third_party_integration/huggingface_transformers/BootstrapNAS_HF/0001-Modifications-for-BootstrapNAS-usage.patch

```diff
@@ -0,0 +1,1918 @@
++    BNASLRSchedulerArgs,
++    update_bnas_lr_scheduler_kwargs,
+     default_data_collator,
+     set_seed,
+ )
+@@ -366,15 +370,6 @@ def main():
+         revision=model_args.model_revision,
+         use_auth_token=True if model_args.use_auth_token else None,
+     )
+-    model = AutoModelForSequenceClassification.from_pretrained(
+-        model_args.model_name_or_path,
+-        from_tf=bool(".ckpt" in model_args.model_name_or_path),
+-        config=config,
+-        cache_dir=model_args.cache_dir,
+-        revision=model_args.model_revision,
+-        use_auth_token=True if model_args.use_auth_token else None,
+-        ignore_mismatched_sizes=model_args.ignore_mismatched_sizes,
+-    )
+ 
+     # Preprocessing the raw_datasets
+     if data_args.task_name is not None:
+@@ -400,12 +395,12 @@ def main():
+     # Some models have set the order of the labels to use, so let's make sure we do use it.
+     label_to_id = None
+     if (
+-        model.config.label2id != PretrainedConfig(num_labels=num_labels).label2id
++        config.label2id != PretrainedConfig(num_labels=num_labels).label2id
+         and data_args.task_name is not None
+         and not is_regression
+     ):
+         # Some have all caps in their config, some don't.
+-        label_name_to_id = {k.lower(): v for k, v in model.config.label2id.items()}
++        label_name_to_id = {k.lower(): v for k, v in config.label2id.items()}
+         if list(sorted(label_name_to_id.keys())) == list(sorted(label_list)):
+             label_to_id = {i: int(label_name_to_id[label_list[i]]) for i in range(num_labels)}
+         else:
+@@ -418,11 +413,11 @@ def main():
+         label_to_id = {v: i for i, v in enumerate(label_list)}
+ 
+     if label_to_id is not None:
+-        model.config.label2id = label_to_id
+-        model.config.id2label = {id: label for label, id in config.label2id.items()}
++        config.label2id = label_to_id
++        config.id2label = {id: label for label, id in config.label2id.items()}
+     elif data_args.task_name is not None and not is_regression:
+-        model.config.label2id = {l: i for i, l in enumerate(label_list)}
+-        model.config.id2label = {id: label for label, id in config.label2id.items()}
++        config.label2id = {l: i for i, l in enumerate(label_list)}
++        config.id2label = {id: label for label, id in config.label2id.items()}
+ 
+     if data_args.max_seq_length > tokenizer.model_max_length:
+         logger.warning(
+@@ -458,6 +453,48 @@ def main():
+             max_train_samples = min(len(train_dataset), data_args.max_train_samples)
+             train_dataset = train_dataset.select(range(max_train_samples))
+ 
++    nncf_config = None
++    if training_args.nncf_config is not None:
++        nncf_config = NNCFConfig.from_json(training_args.nncf_config)
++        nncf_config.register_extra_structs([BNASLRSchedulerArgs(nncf_config)])
++        update_bnas_lr_scheduler_kwargs(nncf_config, training_args)
```

Comment:
With current implementation, `args` registration in the config looks useless, since you immediately use it by extracting from that config.